src/dune_rules/preprocess.ml

```diff
@@ -197,19 +197,18 @@ module Per_module = struct
     else
       No_preprocessing
 
-  let add_instrumentation t ~loc libname =
+  let add_instrumentation t ~loc ~flags:flags' libname =
     Per_module.map t ~f:(fun pp ->
         match pp with
         | No_preprocessing ->
           let pps = [ With_instrumentation.Instrumentation_backend libname ] in
-          let flags = [] in
           let staged = false in
-          Pps { loc; pps; flags; staged }
+          Pps { loc; pps; flags = flags'; staged }
         | Pps { loc; pps; flags; staged } ->
           let pps =
             With_instrumentation.Instrumentation_backend libname :: pps
           in
-          Pps { loc; pps; flags; staged }
+          Pps { loc; pps; flags = flags @ flags'; staged }
```

Comment:
This seems to be wrong when I have a instrumentation backend and  pps at the same time.
In that case, when instrumentation is off, that flags are still passed to pps which caused a build error
@jeremiedimino

Reply:
It indeed looks like we are adding instrumentation flags unconditionally. Thanks @bobzhang for uncovering this and pin-pointing the root of the issue. Do you mind opening an issue about this so that we can track it?

Reply:
See https://github.com/ocaml/dune/pull/4770

Reply:
Fix merged, will be in 2.9

Reply:
Great, thanks @nojb for the prompt fix!